dependency.lic

```diff
@@ -772,257 +773,467 @@ class SlackbotManager
+    settings.hometown = $HOMETOWN if $HOMETOWN
+
+    # If these room settings are defined as maps with a
+    # room id configured per town, then grab the hometown room id
+    # and assign it directly to the setting.
+    [
+      'alchemy_room',
+      'bankbot_room_id',
+      'compost_room',
+      'crossing_training_sorcery_room',
+      'enchanting_room',
+      'engineering_room',
+      'feed_cloak_room',
+      'forage_override_room',
+      'lockpick_room_id',
+      'outdoor_room',
+      'outfitting_room',
+      'prehunt_buffing_room',
+      'safe_room',
+      'safe_room_id',
+      'theurgy_prayer_mat_room'
+    ].each do |setting_name|
+      if settings[setting_name].is_a?(Hash)
+        hometown_value = settings[setting_name][settings.hometown]
+        settings[setting_name] = hometown_value if hometown_value
+      end
+    end
 
     # Merge legacy appraisal settings into target appraisal settings
-    unless s.appraisal_training.include?('pouches')
-      s.appraisal_training.unshift('pouches') if s.train_appraisal_with_pouches
+    unless settings.appraisal_training.include?('pouches')
+      settings.appraisal_training.unshift('pouches') if settings.train_appraisal_with_pouches
     end
-    unless s.appraisal_training.include?('gear')
-      s.appraisal_training.unshift('gear') if s.train_appraisal_with_gear
+    unless settings.appraisal_training.include?('gear')
+      settings.appraisal_training.unshift('gear') if settings.train_appraisal_with_gear
     end
 
     # Merge legacy astrology training settings into target training settings
-    unless s.astrology_training.include?('events')
-      s.astrology_training.unshift('events') if s.predict_event
+    unless settings.astrology_training.include?('events')
+      settings.astrology_training.unshift('events') if settings.predict_event
     end
-    unless s.astrology_training.include?('ways')
-      s.astrology_training.unshift('ways') if s.astral_plane_training['train_in_ap']
+    unless settings.astrology_training.include?('ways')
+      settings.astrology_training.unshift('ways') if settings.astral_plane_training['train_in_ap']
     end
 
-    s
+    return settings
   rescue => e
-    respond('*** ERROR MODIFYING DATA IN DEPENDENCY ***')
-    respond('*** Commonly this is malformed spell or loot entries in your yaml file ***')
-    respond(e.message)
-    e.backtrace.each { |msg| respond(msg) }
-    @settings = {}
-    @load = false
+    echo "*** ERROR TRANSFORMING SETTINGS IN DEPENDENCY ***"
+    echo "*** Commonly this due to malformed config in your yaml file ***"
+    echo e.message
+    e.backtrace.each { |msg| echo msg }
+    OpenStruct.new
   end
+
+  def transform_data(original_data)
+    echo "#{self.class}::#{__callee__}" if @debug
+
+    # Enable dot access to top-level properties (e.g. data.cast_messages)
+    data = OpenStruct.new(original_data)
+
+    # ... reserved for future needs ...
+
+    return data
```

Comment:
The return here is redundant and I believe non idiomatic for ruby.  I don't think the rescue block is anymore vulnerable than any other method layout for someone to come in and drop code after that would alter the return value.